src/snowcli/cli/connection/commands.py

```diff
@@ -194,5 +199,16 @@ def test(connection: str = ConnectionOption, **options) -> CommandResult:
     """
     Tests the connection to Snowflake.
     """
-    connect_to_snowflake(connection_name=connection)
-    return MessageResult("OK")
+    conn = connect_to_snowflake(connection_name=connection)
+    result = {
+        "Connection name": connection,
+        "Status": "OK",
+        "Host": conn.host,
+        "Account": conn.account,
+        "User": conn.user,
+        "Role": f'{conn.role or "not set"}',
+        "Database": f'{conn.database or "not set"}',
+        "Warehouse": f'{conn.warehouse or "not set"}',
+    }
+
+    return ObjectResult(result)
```

Comment:
Should we differentiate status value depending on success/error?

Reply:
success = ok
error = error message

WDYT?

Reply:
Can you give a specific scenario?
I tested it against missing password/username etc. - then, the error output is printed (generated by snow connector)